src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -2396,6 +2396,7 @@ void InterpreterSelectQuery::executeFetchColumns(QueryProcessingStage::Enum proc
             }
             assert(storagestream);
 
+            const String & replay_time_col = settings.replay_time_column;
             if (std::ranges::none_of(required_columns, [](const auto & name) { return name == ProtonConsts::RESERVED_APPEND_TIME; }))
                 required_columns.emplace_back(ProtonConsts::RESERVED_APPEND_TIME);
```

Comment:
use `replay_time_col` instead of prev `_tp_append_time` ?

Reply:
also need validate the replay_time_col is exists in stream (virtual column or physical column)